intel-retail/automated-self-checkout
# feat: Barcode detection and decode

## PR Checklist



## What are you changing?


## Issue this PR will close

close: #**issue_number**

## Anything the reviewer should know when reviewing this PR?

## Test Instructions if applicable

1. Download sample videos again:
cd benchmark-scripts
./download_sample_videos.sh

2. build yolov5 ensemble:
make build-capi_yolov5_ensemble

3. Run camera simulator and pipeline:
make run-camera-simulator
PIPELINE_PROFILE="capi_yolov5_ensemble" RENDER_MODE=1 sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0

4. Go to configs/opencv-ovms/envs/capi_yolov5_ensemble.env and disable BARCODE=0
Re-run the pipeline and barcode should not be detected.

Expected output:

![image (4)](https://github.com/intel-retail/automated-self-checkout/assets/2906855/188e7d0c-0040-44f9-a7d6-8e396557c341)


## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/automated-self-checkout )
